plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java

```diff
@@ -123,7 +123,7 @@ private static String toPredicate(String columnName, Domain domain)
         List<String> disjuncts = new ArrayList<>();
         List<Object> singleValues = new ArrayList<>();
         for (Range range : domain.getValues().getRanges().getOrderedRanges()) {
-            checkState(!range.isAll()); // Already checked
+            checkState(!range.isAll() && !domain.isNullAllowed()); // Already checked
```

Comment:
Let's fix this in a separated PR with some tests.

Reply:
Sure!